src/app/controllers/quotes.js

```
@@ -0,0 +1,97 @@
+const {
+  randomItem,
+  getQuotesByTag,
+  saveQuotes,
+  getID,
+} = require('../lib');
+const quotesData = require('../../../data');
+
+module.exports.getAll = (req, res) => {
+  res.send(quotesData);
+};
+
+module.exports.getRandom = (req, res) => {
+  const { query: { tag } } = req;
+  const quotes = tag ? getQuotesByTag(tag, quotesData) : quotesData;
+
+  if (tag && !quotes.length) {
+    return res.status(404).send('No quotes for provided tag.');
+  }
+
+  res.send(randomItem(quotes));
+};
+
+module.exports.create = (req, res) => {
+  const {
+    author,
+    text,
+  } = req.body;
+
+  if (!author || !text) {
+    return res.status(400).send('"Author" & "text" are required');
+  }
```

It would be great to have it as part of the middleware for the request for the reusability purpose.
See https://expressjs.com/en/guide/using-middleware.html#middleware.router

Also, about fields that are not part of the schema, I think in this case it is better to throw an error.